### ISSUE TITLE
`curly-newline`: minElements option doesn't respect comments

### ISSUE THREAD

**Issue by danyadev** [eslint-stylistic/eslint-stylistic#675]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

If a block contains only comments, the rule tells us that braces should be opened and closed on the same line, which is impossible

```ts
// '@stylistic/curly-newline': ['error', { minElements: 1 }]

// Has problems:
if (someCondition) {
  // The body contains only comments
}

// Perfectly fine, as it should be:
if (someCondition) {
  someFn()
}
```

<img width="740" alt="Image" src="https://github.com/user-attachments/assets/d65fb079-5154-4462-9026-d8065ab96183" />

### Reproduction

https://stackblitz.com/edit/github-qrzehpeq?file=eslint.config.js,src%2Findex.ts

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by keijokapp** [comment]
[`object-curly-newline`](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/packages/eslint-plugin/rules/object-curly-newline/object-curly-newline._js_.ts) (after which this rule has been modeled) also doesn't take comments into account, so I didn't consider this use case when implementing the rule. I'm willing to develop a fix but I don't really know what's the best approach here. Counting comments towards the "elements" would break the rule for those who want single-comment blocks to be on a single line and single-statement blocks on multiple lines. I think there would need to be a new option to specify how comments should be handled but I'm really critical about adding options and complicating the rule configuration.

**Comment by danyadev** [comment]
Does this rule deal with anything related to presence of the curly braces itself?
Or the case you are talking about is like this:
```ts
if (condition) { fn() /* comment */ }
```

because if the rule doesn't care when there are no curly braces:
```ts
if (condition) fn() // another comment (or /* another comment */)
```

it will be a little easier to come up with a solution

the problem is that you can't have single-line body with curly braces and the comments like `// comment`, so we should at least allow multiline body if we have this kind of comments there

and fixing that shouldn't break anything because it only disallows an impossible situation to happen

**Comment by keijokapp** [comment]
The rule only checks code where curly braces are present. I meant the following use case:

```
// allowed
if (condition) { /* comment  */ }

// not allowed
if (condition) {
  /* comment  */
}

// not allowed
if (condition) { foo() }
```

TBH, I don't think disallowing multiline blocks would ever make sense unless the block is completely empty. Maybe the rule should be changed to never disallow non-empty multiline blocks.

**Comment by danyadev** [comment]
I agree with you, so the role of the rule will be to disallow unwanted single-line blocks and inconsistent brace positions

**Comment by waltuov** [comment]
I'd like to fix this. With `minElements: 1` and no `consistent`, a block holding only `// note` has `body.length` 0, and because `first` and `last` are then looked up without comments, `first` is the closing brace itself, so both "Unexpected line break" errors fire with no autofix even though the block can't fit on one line. My plan is to skip those two reports when a comment between the braces makes a single line impossible (a line comment, or a block comment spanning lines), so `if (a) { /* note */ }` keeps working the way keijokapp described. Should the same change also go into `object-curly-newline` for the #971 case, or should this PR stay limited to `curly-newline`? Happy to open a PR with tests.
